src/Markdown/Heading.elm

```diff
@@ -44,15 +44,24 @@ parser =
                             succeed level
                     )
            )
-        |= (Helpers.chompUntilLineEndOrEnd
-                |> Advanced.mapChompedString
-                    (\headingText _ ->
-                        headingText
-                            |> String.trimLeft
-                            |> dropTrailingHashes
-                            |> UnparsedInlines
+        |= oneOf
+            [ succeed (UnparsedInlines "")
+                |. symbol Token.newline
+            , succeed identity
+                |. oneOf
+                    [ symbol Token.space
+                    , symbol Token.tab]
+                |= (Helpers.chompUntilLineEndOrEnd
+                        |> Advanced.mapChompedString
+                            (\headingText _ ->
+                                headingText
+                                    |> String.trimLeft
+                                    |> String.trimRight
```

Comment:
```suggestion
                                    |> String.trim
```

This should be equivalent to `trim` (and might be a little bit more performant possibly).